src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/TypeNameCheck.java

```diff
@@ -63,29 +63,53 @@
  * An example of how to configure the check is:
  * </p>
  * <pre>
- * &lt;module name="TypeName"/&gt;
+ * &lt;module name=&quot;TypeName&quot;/&gt;
+ * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * public interface FirstName {} // OK
+ * protected class SecondName {} // OK
+ * enum Third_Name {} // violation, name 'Third_Name' must match pattern '^[A-Z][a-zA-Z0-9]*$'
+ * private class 4thName {} // violation, name '4thName' must match pattern '^[A-Z][a-zA-Z0-9]*$'
```

Comment:
`4thName`
This class name is invalid and won't compile, should be changed.

Reply:
Done